test/mri/tests/io/wait/test_io_wait.rb

```diff
@@ -161,6 +161,34 @@ def test_wait_readwrite_timeout
     assert_equal @w, @w.wait(0.01, :read_write)
   end
 
+  def test_wait_mask_writable
+    omit("Missing IO::WRITABLE!") unless IO.const_defined?(:WRITABLE)
+    assert_equal IO::WRITABLE, @w.wait(IO::WRITABLE, 0)
+  end
+
+  def test_wait_mask_readable
+    omit("Missing IO::READABLE!") unless IO.const_defined?(:READABLE)
+    @w.write("Hello World\n" * 3)
+    assert_equal IO::READABLE, @r.wait(IO::READABLE, 0)
+
+    @r.gets
+    assert_equal IO::READABLE, @r.wait(IO::READABLE, 0)
+  end
+
+  def test_wait_mask_zero
+    omit("Missing IO::WRITABLE!") unless IO.const_defined?(:WRITABLE)
+    assert_raise(ArgumentError) do
+      @w.wait(0, 0)
+    end
+  end
+
+  def test_wait_mask_negative
+    omit("Missing IO::WRITABLE!") unless IO.const_defined?(:WRITABLE)
+    assert_raise(ArgumentError) do
+      @w.wait(-6, 0)
+    end
+  end
```

Comment:
We avoid modifying MRI tests since that can cause conflict e.g. when updating them for 3.2 support.
From which CRuby version or io-wait gem version do these tests come from?

Reply:
From ruby master branch here: https://github.com/ruby/ruby/blob/44f444478afad954f046d8345cafb3842708a4ca/test/io/wait/test_io_wait.rb#L164-L190

Reply:
> but it seems that they don't run in CI currently.

They do but only in the internal CI. MRI tests are quite slow, i.e. too slow to run on GitHub Actions.

Reply:
Let's merge these for additional coverage.
In general we prefer ruby/spec specs as those are typically better written (e.g., these tests are just skip if IO::READABLE is missing, that's not good behavior for testing alternative Ruby implementations).